### DISCUSSION TITLE
Setting Charging times for FoxESS from HA

### DISCUSSION THREAD

**Discussion by EricMac1** [macxq/foxess-ha#102] [Q&A]
Is it possible to use HomeAssistant (HA) to set the Battery charging periods from the grid? I have installed HA with the FoxESS integration using the Ethernet cable and it is monitoring the system well. I am looking to be able to fully control the inverter without the need for using the FoxESS cloud service as I find that the system goes down more often than I would like.......

**Comment by rawlingsonline1** [comment]
hi Eric did you ever work out how to do this?

**Reply by EricMac1** [reply]
Hi. No I am still using the Foxcloud App to adjust the Feed operating mode & charging periods. The App has been a lot more stable recently so has not caused any issues. I am aware that there may be an integration which can be used with HA to adjust the charge times but I have not looked in detail. But when I have looked previously the interface remains with the Foxcloud system.

My system is a lot more stable now and manual intervention minimal since the Global Fox Applications Engineering Manager got involved and reprogrammed the BMS firmware. But I am looking to re-evaluate my HA integration over the next month or so as there has been significant progress for monitoring & making adjustments using HA for the FoxESS systems.

**Reply by Daibutt** [reply]
If you are connecting to inverter using modbus (either Lan or RS485) then this is worth checking out;
https://github.com/nathanmarlor/foxess_modbus

**Comment by rawlingsonline1** [comment]
It's via cloud right now as only been using home assistant a week and still
very much learning. Thanks for the info I definitely will look into it and
give it a try.

**Reply by Daibutt** [reply]
Don't know if you've come across these but....

Info on different ways to get your data (but the integration I posted above isn't listed as it's newer, but this gives you some pros/cons)
https://foxesscommunity.com/viewtopic.php?t=18

And these for just info/discussion;

https://www.facebook.com/groups/foxessownersgroup
https://foxesscommunity.com/
:)
